### PR TITLE
fix some typo error

### PR DESCRIPTION
**Pull request by PanternBao**
### Task Description
correct some source file typos.

### Solution Description
correct spell.

### Passed Regressions
compile successfully

### Upgrade Compatibility
No

### Other Information
No

### Release Note
No

### CODE REVIEW

File: src/sql/code_generator/ob_static_engine_cg.cpp [hunk 1]
```diff
@@ -2733,7 +2733,7 @@ int ObStaticEngineCG::generate_spec(ObLogGranuleIterator &op, ObGranuleIteratorS
       "desc", op.desc_order(),
       "flags", op.get_gi_flags(),
       "tsc_ids", spec.pw_dml_tsc_ids_,
-      "repart_prunint_idx", spec.repart_pruning_tsc_idx_,
+      "repart_pruning_idx", spec.repart_pruning_tsc_idx_,
```
Comment (akaError) [inline src/sql/code_generator/ob_static_engine_cg.cpp:2736]: 'repart_pruning_tsc_idx" would be better， the original "pruning" means "pruning_tsc"

File: src/sql/code_generator/ob_static_engine_expr_cg.cpp [hunk 2]
```diff
@@ -614,8 +614,8 @@ int ObStaticEngineExprCG::cg_param_frame_layout(const ObIArray<ObRawExpr *> &par
         rt_expr->res_buf_off_ = 0;
         rt_expr->res_buf_len_ = 0;
         // 对于T_QUESTIONMARK的param表达式, 使用extra_记录其实际value在
-        // datam_param_store中下标,  通过该下标, 在执行期可以访问
-        // datum_param_store中存放的信息,
+        // data_param_store中下标,  通过该下标, 在执行期可以访问
```
Comment (akaError) [inline src/sql/code_generator/ob_static_engine_expr_cg.cpp:617]: "datum" is correct, need no change

File: src/observer/virtual_table/ob_show_create_table.cpp [hunk 3]
```diff
@@ -213,7 +213,7 @@ int ObShowCreateTable::fill_row_cells_inner(const uint64_t show_table_id,
           ObSchemaPrinter schema_printer(*schema_guard_, strict_mode, sql_quote_show_create, ansi_quotes);
           int64_t pos = 0;
           if (table_schema.is_view_table()) {
-            if (OB_FAIL(schema_printer.print_view_definiton(effective_tenant_id_,
+            if (OB_FAIL(schema_printer.print_view_definition(effective_tenant_id_,
```
[the remark this answers is not in the document]
Comment (PanternBao) [reply src/observer/virtual_table/ob_show_create_table.cpp:216]: The code has been compiled successfully, and I only change structure or function name, so
 the code doesn't affect function. in this case,do I need to remove this part of the code too?

[the remark this answers is not in the document]
Comment (akaError) [reply src/observer/virtual_table/ob_show_create_table.cpp:216]: yes, please remove it

File: src/sql/plan_cache/ob_plan_cache_value.cpp [hunk 4]
```diff
@@ -902,7 +902,7 @@ int ObPlanCacheValue::resolve_insert_multi_values_param(ObPlanCacheCtx &pc_ctx,
                                        must_be_positive_idx,
                                        *raw_param_array,
                                        &temp_obj_params))) {
-      LOG_WARN("failed to resolve parames", K(ret));
+      LOG_WARN("failed to resolve params", K(ret));
```
Comment (akaError) [inline src/sql/plan_cache/ob_plan_cache_value.cpp:905]: error: patch failed: src/sql/code_generator/ob_dml_cg_service.cpp
error: patch failed: src/sql/ob_select_stmt_printer.cpp
error: patch failed: src/sql/plan_cache/ob_plan_cache_value.cpp

remove the modification of the above three file, there is conflict with internal codebase

**Pull request merged**